cdap-ui/app/cdap/components/Wizard/index.js

```diff
@@ -104,186 +101,121 @@ export default class Wizard extends Component {
     this.setState({loading: true});
     if (onSubmitReturn instanceof Rx.Observable) {
       onSubmitReturn
-        .subscribe(
-          () => {
+      .subscribe(
+        () => {
+          this.setState({
+            error: false,
+            loading: false
+          });
+          if (this.props.successInfo) {
+            this.setState({callToActionInfo: this.props.successInfo});
+          } else {
+            this.props.onClose(true);
+          }
+        },
+        (err) => {
+          if (err) {
             this.setState({
-              success: true,
-              successInfo: this.props.successInfo,
-              error: false,
+              error: err,
               loading: false
             });
-          },
-          err => {
-            if (err) {
-              this.setState({
-                error: err,
-                success: false,
-                loading: false
-              });
-            }
           }
-        );
+        }
+      );
     }
   }
-  render() {
-    const getNavigationButtons = function getNavigationButtons(matchedStep) {
-      let matchedIndex = currentStepIndex(this.props.wizardConfig.steps, matchedStep.id);
-      let navButtons;
-      let nextButton = (
-        <button
-          className="btn btn-secondary"
-          onClick={this.goToNextStep.bind(this, matchedStep.id)}
-        >
-          <span>Next</span>
-          <span className="fa fa-chevron-right"></span>
-        </button>
+  isStepComplete(stepId) {
+    let state = this.props.store.getState()[stepId];
+    return state && state.__complete;
+  }
+
+  getNavigationButtons(matchedStep) {
+    let matchedIndex = currentStepIndex(this.props.wizardConfig.steps, matchedStep.id);
+    let navButtons;
+    let nextButton = (
+      <button
+        className="btn btn-secondary"
+        onClick={this.goToNextStep.bind(this, matchedStep.id)}
+      >
+        <span>Next</span>
+        <span className="fa fa-chevron-right"></span>
+      </button>
+    );
+    let prevButton = (
+      <button
+        className="btn btn-secondary"
+        onClick={this.goToPreviousStep.bind(this, matchedStep.id)}
+      >
+        <span className="fa fa-chevron-left"></span>
+        <span>Previous</span>
+      </button>
+    );
+    let finishButton = (
+      <button
+        className="btn btn-primary"
+        onClick={this.submitForm.bind(this)}
+        disabled={(!this.state.requiredStepsCompleted || this.state.loading) ? 'disabled' : null}
+      >
+        Finish
+      </button>
+    );
+    // This is ugly. We need to find a better way.
```

Comment:
This part is about figuring out which step in a particular Wizard the `Finish` button should appear at, so I don't think it's directly relevant here.